Sources/BraintreePayPal/BTPayPalVaultRequest.swift

```diff
@@ -19,6 +19,9 @@ import BraintreeCore
     /// - Warning: This property is currently in beta and may change or be removed in future releases.
     var enablePayPalAppSwitch: Bool = false
 
+    /// exposed for mocking in tests
+    var application: URLOpener = UIApplication.shared
```

Comment:
IMO it's a little strange for VaultRequest to do the app installed inspection. It adds logic into our "model" type

Reply:
Agree, but it also felt odd that we were overriding a property in the client to ensure we don't pass the app switch params if the app is not installed here: https://github.com/braintree/braintree_ios/blob/6a375f2b7733e685e31cb0f1cd4d8909208ba4fb/Sources/BraintreePayPal/BTPayPalClient.swift#L336-L338

Do you know where else this logic could live that would make more sense? We could potentially pass it as a property when we call `parameters()` similar to `universalLink`. Curious to hear your thoughts.

Reply:
IMO the way it exists now is the highest separation of concerns (business logic in PayPalClient, model only in VaultRequest)

Reply:
I understand the concern now - currently we're changing the merchant-facing `VaultRequest. enablePayPalAppSwitch` setting without them knowing, which could be confusing. Jax I like the idea of injecting into the `parameters()` method more than having a reference to `UIApplication` in the `VaultRequest`!

Reply:
Yeah exactly - sorry I realize I should have done a better job of explaining the intent of the change initially. Pushed the change here: https://github.com/braintree/braintree_ios/pull/1381/commits/893d7363221beaa8711f453415fd6249d22d43bf